build.gradle.kts

```diff
@@ -0,0 +1,121 @@
+plugins {
+    alias(libs.plugins.kotlin.jvm) apply false
+    alias(libs.plugins.checksum)
+    alias(libs.plugins.shadow)
+    alias(libs.plugins.githubRelease)
+}
+
+val isKotlinDev: Boolean = project.hasProperty("isKotlinDev")
+
+allprojects {
+    if (isKotlinDev) {
+        val definedVersion = ext["VERSION_NAME"].toString().removeSuffix("-SNAPSHOT")
+        ext["VERSION_NAME"] = "$definedVersion-kotlin-dev-SNAPSHOT"
+    }
+
+    tasks.withType<Test>().configureEach {
+        useJUnitPlatform()
+    }
+}
+
+val ktlint: Configuration = configurations.create("ktlint")
+
+dependencies {
+    ktlint(projects.ktlint)
+}
+
+tasks.register<JavaExec>("ktlint") {
+    group = LifecycleBasePlugin.VERIFICATION_GROUP
+    description = "Check Kotlin code style including experimental rules."
+    classpath = ktlint
+    mainClass.set("com.pinterest.ktlint.Main")
+    args(
+        "**/src/**/*.kt",
+        "**.kts",
+        "!**/build/**",
+        // Exclude sources which contain lint violations for the purpose of testing.
+        "!ktlint/src/test/resources/**",
+        "--baseline=ktlint/src/test/resources/test-baseline.xml",
+        // Experimental rules run by default run on the ktlint code base itself. Experimental rules should not be released if
+        // we are not pleased ourselves with the results on the ktlint code base.
+        "--experimental",
+        // Do not run with option "--verbose" or "-v" as the lint violations are difficult to spot between the amount of
+        // debug output lines.
+    )
+}
+
+// Deployment tasks
+val githubToken: String = if (project.hasProperty("servers.github.privKey")) {
+    project.property("servers.github.privKey").toString()
+} else {
+    logger.warn("No github token specified")
+    ""
+}
+
+val shadowJarExecutable: TaskProvider<Task> by lazy {
+    projects.ktlint.dependencyProject.tasks.named("shadowJarExecutable")
+}
+
+// Explicitly adding dependency on "shadowJarExecutable" as Gradle does not it set via "releaseAssets" property
+tasks.named("githubRelease") {
+    dependsOn(provider { shadowJarExecutable })
+}
```

Comment:
Not very sure here, but guessing the previously publish failure was related to

https://github.com/pinterest/ktlint/blob/cdc39b12b215fc76205fd02fe69ccfe8f77d6d81/build.gradle.kts#L60-L64